### PR TITLE
add composer version note

### LINKED ISSUE #3890: Drupal Commerce on Drupal 8 Doc Update
Re: [Drupal Commerce on Drupal 8](https://pantheon.io/docs/guides/drupal-8-commerce/)

Priority: Medium

## Issue Description:

Couldnt get it installed with:

composer require "drupalcommerce/commerce_base dev-8.x-1.x"

changed to:

composer require 'drupal/commerce:^2.8'

Although it still kicks a bunch of these back:

strpos(): Empty needle ExtensionDiscovery.php:295                      [warning]
The following module is missing from the file system: commerce         [warning]
bootstrap.inc:268

Also, in the Install Commerce section item 6, I didnt get the Install Later option and the Drop table notificatoins had me nervous.

## Suggested Resolution

Change the composer require line, remove the Install Later note and let people know its ok to drop the table.

### LINKED ISSUE #4517: Convert a Standard Drupal 8 Site to a Composer Managed Site Doc Update – composer require fails
Re: [Convert a Standard Drupal 8 Site to a Composer Managed Site](https://pantheon.io/docs/guides/composer-convert/)

Priority: Medium

## Issue Description:
This guide tells me to add existing contrib modules to the new codebase using the composer require command. However when I try to run this command with `drupal/admin_toolbar` on a fresh `pantheon-systems/example-drops-8-composer` project, I get the following errors from composer:

> Your requirements could not be resolved to an installable set of packages.
> 
>   Problem 1
>     - The requested package drush-ops/behat-drush-endpoint (locked at 0.0.5, required as ^8) is satisfiable by drush-ops/behat-drush-endpoint[0.0.5] but these conflict with your requirements or minimum-stability.
> 
> 
> Installation failed, reverting ./composer.json to its original content.

### Steps to reproduce

1. `composer create-project pantheon-systems/example-drops-8-composer pantheon-composer`
2. `cd pantheon-composer`
3. `composer require drupal/admin_toolbar`

## Suggested Resolution
After removing the composer.lock file and the vendor folder, composer require runs successfully.

### PR DESCRIPTION
**Pull request by alexfornuto**
~Closes~ Need raised by #4517 
~Closes~ Need raised by #3890

## Effect
PR includes the following changes:
- Creates a new partial file with a note describing how composer lock files may need to be updates as software dependencies and versioning is updated.
- Adds the warning to a couple docs.

## Remaining Work
- [x] Technical review of the note itself (@dustinleblanc ?)
- [x] Confirm it relates to both issues referenced above (@stevector ? )
- [x] Check for additional composer-based docs this should be added to.
- [x] Copy review

## Post Launch
To be completed by the docs team upon merge: 
- [ ] Update Status Report
- [ ] Archive from **Done** in Waffle

### CODE REVIEW

File: source/_partials/content/composer-updating.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div class="alert alert-info" role="alert">
+  <h4 class="info">Note</h4>
+  <p>As packages pulled by Composer are updated (along with their dependencies), version compatibility issues can pop up. `composer.lock` prevents the updating of packages beyond the version limits specified within, and need to be manually reviewed/updated from time to time. See the <a href="https://getcomposer.org/doc/01-basic-usage.md#updating-dependencies-to-their-latest-versions" class="external">updating dependencies</a> section of Composer's documentation for more information.</p>
```
Comment (stevector) [inline source/_partials/content/composer-updating.html]: composer.lock is meant to stop an version changes unless you updating (not just those beyond the limits in composer.json)
```suggestion
  <p>As packages pulled by Composer are updated (along with their dependencies), version compatibility issues can pop up. Sometimes you may need to manually alter the version constraints on a given package within the `require` or `require-dev` section of `composer.json` in order to update packages. See the <a href="https://getcomposer.org/doc/01-basic-usage.md#updating-dependencies-to-their-latest-versions" class="external">updating dependencies</a> section of Composer's documentation for more information.</p>
```

**Comment by stevector** [comment]
I added a suggested rewording. I think it helps resolve both related issues. But both issues could/should also be resolved more directly. @alexfornuto I let's review in person next week.

**Comment by alexfornuto** [comment]
@stevector please take a look at the latest commit for accuracy.

File: source/_partials/content/composer-updating.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="alert alert-info" role="alert">
+  <h4 class="info">Note</h4>
+  <p>As packages pulled by Composer are updated (along with their dependencies), version compatibility issues can pop up. Sometimes you may need to manually alter the version constraints on a given package within the <code>require</code> or <code>require-dev</code> section of <code>composer.json</code> in order to update packages. See the <a href="https://getcomposer.org/doc/01-basic-usage.md#updating-dependencies-to-their-latest-versions" class="external">updating dependencies</a> section of Composer's documentation for more information.</p>
+  <br>
+  <p>As a first troubleshooting step, try running <code>composer update</code> to bring <code>composer.lock</code> up to date with <code>composer.json</code>.</p>
```
Comment (stevector) [inline source/_partials/content/composer-updating.html]:
```suggestion
  <p>As a first troubleshooting step, try running <code>composer update</code> to bring <code>composer.lock</code> up to date with the latest available packages.</p>
```

**Comment by stevector** [comment]
@alexfornuto, suggestion added!

File: source/_partials/content/composer-updating.html (hunk 2 above)
Comment (alexfornuto) [reply source/_partials/content/composer-updating.html]: Thanks!

**Pull request merged**